libraries/provider_helpers.rb

```diff
@@ -69,21 +85,23 @@ def validate_extracted_app
       end
 
       def validate_versions # rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+        unless new_cache_path.exist?
+          raise "Downloaded app is not structured correctly. Expected sub-directory named #{name}, got #{(app_cache_path + 'new').children(false)}"
+        end
+
         app_version = version
         pkg_app_conf = CernerSplunk::ConfHelpers.read_config(new_cache_path + 'default/app.conf')
-        return true unless app_version && pkg_app_conf.key?('launcher')
+
+        # If the version is not specified, or the version does not exist in the currently installed app, then continue to install.
+        return true unless app_version && pkg_app_conf.key?('launcher') && pkg_app_conf.dig('launcher', 'version')
```

Comment:
This doesn't line up with the comment... This is checking the desired version and the packaged app version... not the installed app version...

Reply:
There were actually a few errors in this validation...
84ed597deb78a3d626dfa879b5d2f40bca2c1ebe

I added a check to validate_packaged_app to ensure the downloaded app has a version, removed a redundant and untested check for the downloaded app directory (validate_packaged_app already does this), and changed this to check that the current desired version and current installed version exist